modules/consensus/synchronize.go

```diff
@@ -94,24 +113,24 @@ func (s *State) sendBlocks(conn modules.PeerConn) error {
 		return err
 	}
 
-	// Find the most recent block from knownBlocks that is in our current path.
-	id := s.mu.RLock()
+	// Find the most recent block from knownBlocks in the current path.
 	found := false
 	var start types.BlockHeight
-	for _, id := range knownBlocks {
-		bn, exists := s.blockMap[id]
-		if exists && bn.height <= s.height() && id == s.currentPath[bn.height] {
-			found = true
-			start = bn.height + 1 // start at child
-			break
+	lockID := s.mu.RLock()
+	{
+		for _, id := range knownBlocks {
+			bn, exists := s.blockMap[id]
+			if exists && bn.height <= s.height() && id == s.currentPath[bn.height] {
+				found = true
+				start = bn.height + 1 // start at child
+				break
+			}
```

Comment:
hmm. Maybe. It just felt like a lot of code was separating the two, I didn't see the for block when looking at it.

I also moved the variables some, perhaps that helped. Maybe the extra scoping isn't needed.